app/controllers/shopping_list_controller.rb

```diff
@@ -0,0 +1,51 @@
+class ShoppingListController < ApplicationController
+  def index
+    @food_amount = 0
+    @total_price = 0
+    @recipe_foods = []
+    @need_foods = RecipeFood.where(recipe_id: params[:recipe_id])
+    @need_foods.each do |need_food|
+      new_food_id = need_food.food_id
+      new_val = Food.where(user: current_user).where(id: new_food_id).first.quantity
+      quantity_needed = new_val - need_food.quantity
+      next if quantity_needed >= 0
+
+      need_food.quantity = quantity_needed * -1
+      @recipe_foods << need_food
+    end
+    @recipe_foods.each do |f|
+      @total_price += f.food.price * f.quantity
+    end
```

Comment:
- [ ] I created two recipes (not public), and add two foods, then it was time to generate a shopping list. When only one food is added to a recipe, the shopping list doesn't record anything. When I add another food to a recipe, it is recorded as the only one (ingredient) available. This issue doesn't conform to the project requirements. Kindly make changes so that this issue is addressed.

![shopping list](https://user-images.githubusercontent.com/53124548/227473952-4a987993-006a-454d-8017-9a430f977a9a.png)

![one food](https://user-images.githubusercontent.com/53124548/227473999-f08a5198-f50c-4030-a6f6-32312881602e.png)